### PR TITLE
Special views 

### LINKED ISSUE #862: Special view for answers
Given a section, we should have a view showing all answers (for no staff user)

### PR DESCRIPTION
**Pull request by sonlinux**
fix #862 
View that shows correct answers to corresponding worksheet questions.

*On detail page*
spatial

### CODE REVIEW

File: django_project/lesson/views/worksheet.py [hunk 1]
```diff
@@ -257,3 +260,48 @@ def post(self, request, *args, **kwargs):
...
+    """
+    # headline = _(u'Worksheet answers to module questions')
+    context_object_name = 'worksheets'
+    template_name = 'worksheet/question_answers.html'
+
+    def get_context_data(self, **kwargs):
+        """Create context for use in the templates."""
+
+        context = super(WorksheetModuleQuestionAnswers, self).get_context_data(
+         **kwargs)
```
Comment (ismailsunni) [inline django_project/lesson/views/worksheet.py]: __[Flake8]__

[E121] continuation line under-indented for hanging indent

File: django_project/lesson/views/worksheet.py [hunk 2]
```diff
@@ -257,3 +260,42 @@ def post(self, request, *args, **kwargs):
...
+    # headline = _(u'Worksheet answers to module questions')
+    context_object_name = 'worksheets'
+    template_name = 'worksheet/question_answers.html'
+
+    def get_context_data(self, **kwargs):
+        """Create context for use in the templates."""
+
+        context = super(WorksheetModuleQuestionAnswers,
+                        self).get_context_data(**kwargs)
+        pk = self.kwargs.get('pk', None)
```
Comment (ismailsunni) [inline django_project/lesson/views/worksheet.py]: __[Flake8]__

[F841] local variable 'pk' is assigned to but never used

File: django_project/lesson/views/worksheet.py [hunk 3]
```diff
@@ -257,3 +260,50 @@ def post(self, request, *args, **kwargs):
...
+                worksheet_json = {'worksheet': worksheet,
+                                   'question_answers':[]
+                                  }
+
+                worksheet_questions = WorksheetQuestion.objects.filter(
+                        worksheet=worksheet.pk)
+
+                for question in worksheet_questions:
+                    question_json = {'question':
+                                         question,
```
Comment (ismailsunni) [inline django_project/lesson/views/worksheet.py]: __[Flake8]__

[E127] continuation line over-indented for visual indent

File: django_project/lesson/views/worksheet.py [hunk 4]
```diff
@@ -257,3 +260,50 @@ def post(self, request, *args, **kwargs):
...
+    :type request: HttpRequest
+    """
+    context_object_name = 'worksheets'
+    template_name = 'worksheet/question_answers.html'
+
+    def get_context_data(self, **kwargs):
+        """Create context for use in the templates."""
+
+        context = super(WorksheetModuleQuestionAnswers, self).get_context_data(
+         **kwargs)
```
Comment (ismailsunni) [inline django_project/lesson/views/worksheet.py]: __[Flake8]__

[E121] continuation line under-indented for hanging indent

File: django_project/lesson/views/worksheet.py [hunk 5]
```diff
@@ -257,3 +260,50 @@ def post(self, request, *args, **kwargs):
...
+
+        context['sections'] = Section.objects.filter(project=project,
+                                                     slug=section_slug)
+        for section in context['sections']:
+            query_set = Worksheet.objects.filter(section=section)
+            context['worksheets'] = []
+
+            for worksheet in query_set:
+                worksheet_json = {'worksheet': worksheet,
+                                   'question_answers':[]
```
Comment (ismailsunni) [inline django_project/lesson/views/worksheet.py]: __[Flake8]__

[E127] continuation line over-indented for visual indent

Comment (ismailsunni) [inline django_project/lesson/views/worksheet.py]: __[Flake8]__

[E231] missing whitespace after ':'

**Comment by ismailsunni** [comment]
Flake8 detected 8 issues on c1a5f96c735f913c207b3893ec72f4990efe8e3d
Visit https://sideci.com/gh/10589087/pull_requests/915 to review the issues.

**Comment by ismailsunni** [comment]
Flake8 detected 7 issues on b3d20266658383324ae3c67d7221e7b0bc626de1
Visit https://sideci.com/gh/10589087/pull_requests/915 to review the issues.

**Comment by timlinux** [comment]
Please make your tests pass so we can merge @sonlinux

File: django_project/lesson/tests/test_worksheet_views.py [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+                    'section_slug': self.test_section.slug}),
+                post_data)
+        self.assertEqual(response.status_code, 302)
+
+    @override_settings(VALID_DOMAIN = ['testserver', ])
+    def test_WorksheetCreateView_with_login(self):
+        """Test accessing worksheet create view with login."""
+
+        status = self.client.login(username = 'sonlinux', password =
+        'password')
```
Comment (ismailsunni) [inline django_project/lesson/tests/test_worksheet_views.py]: __[Flake8]__

[E128] continuation line under-indented for visual indent

File: django_project/lesson/tests/test_worksheet_views.py [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+
+    @override_settings(VALID_DOMAIN = ['testserver', ])
+    def test_WorksheetModuleQuestionAnswers_with_login(self):
+
+        status = self.client.login(username = 'sonlinux',
+                                   password = 'password')
+        self.assertTrue(status)
+
+        response = self.client.get(reverse('worksheet-module-answers', kwargs =
+        self.kwargs_worksheet_full))
```
Comment (ismailsunni) [inline django_project/lesson/tests/test_worksheet_views.py]: __[Flake8]__

[E128] continuation line under-indented for visual indent

**Pull request merged**